src/subpackages/library/cellprofiler_library/functions/measurement.py

```diff
@@ -2992,3 +3000,537 @@ def make_objskeleton_graph(
+    first_object_index = order[:, first_neighbor]
+    first_x_vector = ncenters[first_object_index, 1] - ocenters[:, 1]
+    first_y_vector = ncenters[first_object_index, 0] - ocenters[:, 0]
+    if nneighbors > first_neighbor + 1:
+        second_neighbor = first_neighbor + 1
+        second_object_index = order[:, second_neighbor]
+        second_x_vector = ncenters[second_object_index, 1] - ocenters[:, 1]
+        second_y_vector = ncenters[second_object_index, 0] - ocenters[:, 0]
+        v1 = numpy.array((first_x_vector, first_y_vector))
+        v2 = numpy.array((second_x_vector, second_y_vector))
+        #
+        # Project the unit vector v1 against the unit vector v2
+        #
+        dot = numpy.sum(v1 * v2, 0) / numpy.sqrt(
+            numpy.sum(v1 ** 2, 0) * numpy.sum(v2 ** 2, 0)
+        )
+        angle = numpy.arccos(dot) * 180.0 / numpy.pi
+    first_x_vector = first_x_vector[object_indexes]
+    second_x_vector = second_x_vector[object_indexes]
+    first_y_vector = first_y_vector[object_indexes]
+    second_y_vector = second_y_vector[object_indexes]
+    angle = angle[object_indexes]
+    first_closest_distance = numpy.sqrt(first_x_vector ** 2 + first_y_vector ** 2)
+    second_closest_distance = numpy.sqrt(second_x_vector ** 2 + second_y_vector ** 2)
+    return (
+        first_closest_distance,
+        second_closest_distance,
+        angle,
+    )
+
+def get_extended_dilated_patch(
+        patch_mask: NDArray[numpy.bool_], 
+        strel:  NDArray[numpy.float_], 
+        distance: int
+    ):
+    if distance <= 5:
+        extended = scipy.ndimage.binary_dilation(patch_mask, strel)
+    else:
+        extended = (scipy.signal.fftconvolve(patch_mask, strel, mode="same") > 0.5)
+    return extended
+
+def measure_object_neighbors(
+        objects_small_removed_segmented: ObjectSegmentation, 
+        kept_labels: ObjectSegmentation,
+        neighbor_small_removed_segmented: ObjectSegmentation, 
+        neighbor_kept_labels: ObjectSegmentation,
+        neighbors_are_objects: bool,
+        dimensions: int, 
+        distance_value:int, 
+        distance_method: NeighborsDistanceMethod, 
+        wants_excluded_objects: bool=True,
+        ) -> Tuple[
+        NDArray[numpy.float_],
+        NDArray[numpy.int_],
+        NDArray[numpy.int_],
+        NDArray[numpy.float_],
+        NDArray[numpy.float_],
+        NDArray[numpy.float_],
+        NDArray[numpy.float_],
+        NDArray[numpy.int_],
+        NDArray[numpy.int_],
+        Optional[NDArray[numpy.int_]],
+    ]:
+    labels: ObjectSegmentation = objects_small_removed_segmented.copy()
+    neighbor_labels: ObjectSegmentation = neighbor_small_removed_segmented.copy()
+    
+    kept_label_set = cast_labels_to_label_set(kept_labels)
+    neighbor_kept_label_set = cast_labels_to_label_set(neighbor_kept_labels)
+    kept_label_ijv = convert_label_set_to_ijv(kept_label_set, validate=False)
+    has_pixels = areas_from_ijv(kept_label_ijv) > 0
```

Comment:
These functions from functions/segmented can be expensive. Library user should be able to pass these as args in order to avoid having to re-compute these values